extensions/applicationinsights-dependencies-js/src/ajax.ts

```diff
@@ -65,6 +66,27 @@ function _supportsAjaxMonitoring(): boolean {
         result = false;
     }
 
+    if (result) {
+        // Disable if the XmlHttpRequest can't be extended or hooked
+        try {
+            let xhr = new XMLHttpRequest();
+            xhr[strAjaxData] = {};
+
+            // Check that we can update the prototype
+            let theOpen = XMLHttpRequest[strPrototype].open;
+            XMLHttpRequest[strPrototype].open = theOpen;
+        } catch (e) {
+            // We can't decorate the xhr object so disable monitoring
```

Comment:
When/how would prototype object be locked down?

Reply:
The prototype is just an object, so it can also be locked down using the Object.preventExtensions(); Reflect.preventExtensions() and Object.freeze() directly on the XMLHttpRequest.prototype object.